### PR TITLE
add check arg to regenerate

### PR DESCRIPTION
**Pull request by CJ-Wright**
@scopatz I think this will help with our rever check

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/cli.py [hunk 1]
```diff
@@ -342,6 +342,7 @@ def __init__(self, parser):
             help="Exclusive conda-build config file to replace conda-forge-pinning. "
             + "For advanced usage only",
         )
+        scp.add_argument("--check", action="store_false", help="Check if regenerate can be performed")
```
Comment (scopatz) [inline conda_smithy/cli.py]: Shouldn't this be `action="store_true", default=False`?

**Comment by scopatz** [comment]
Thanks! I think this needs a news entry, though

**Comment by CJ-Wright** [comment]
@scopatz ready for another round

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**